### PR TITLE
Added "send dimensions" button

### PR DESCRIPTION
**Pull request by LonicaMewinsky**
Small UI addition -- sends the dimensions of a particular ControlNet tab's **current image** to txt2img or img2img. Relates to https://github.com/Mikubill/sd-webui-controlnet/issues/378.

[Franco.webm](https://user-images.githubusercontent.com/93007558/221983147-d9ee91ac-5e74-48f6-940d-71e739f03c8d.webm)

### CODE REVIEW

File: scripts/controlnet.py [hunk 1]
```diff
@@ -230,7 +250,20 @@ def uigroup(self, is_img2img):
...
+        def send_dimensions(image):
+            def closesteight(num):
+                rem = num % 8
+                if rem <= 4:
+                    return round(num - rem)
+                else:
+                    return round(num + (8 - rem))
+            if(image):
+                interm = np.asarray(image.get('image'))
+                return gr.Slider.update(value=closesteight(interm.shape[1])), gr.Slider.update(value=closesteight(interm.shape[0]))
```
Comment (missionfloyd) [inline scripts/controlnet.py:263]:
```suggestion
                return closesteight(interm.shape[1]), closesteight(interm.shape[0])
```
[update()](https://www.gradio.app/docs/#update-header) is only necessary to set properties other than value.

File: scripts/controlnet.py [hunk 2]
```diff
@@ -230,7 +250,20 @@ def uigroup(self, is_img2img):
...
+            else:
+                return gr.Slider.update(), gr.Slider.update()
```
Comment (missionfloyd) [inline scripts/controlnet.py:265]:
```suggestion
                return closesteight(interm.shape[1]), losesteight(interm.shape[0])
```
[update()](https://www.gradio.app/docs/#update-header) is only necessary to set properties other than value.

File: scripts/controlnet.py (hunk 1 above)
Comment (missionfloyd) [inline scripts/controlnet.py:263]:
```suggestion
                return closesteight(interm.shape[1]), closesteight(interm.shape[0])
```

Comment (LonicaMewinsky) [reply scripts/controlnet.py]: Heck yea. Thanks.

**Comment by GeorgLegato** [comment]
it is included in my PR as well (and lot of more stuff: tooltips, proportional scale tool, quick buttons 512/768):
#336

**Pull request merged**